Purchases/Caching/RCInMemoryCachedObject.m

```diff
@@ -0,0 +1,70 @@
+//
+//  RCInMemoryCachedObject.m
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+
+#import "RCInMemoryCachedObject.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface RCInMemoryCachedObject ()
+
+@property (nonatomic, nullable) NSDate *lastUpdatedAt;
+@property (nonatomic, assign) int cacheDurationInSeconds;
+@property (nonatomic, nullable) id cachedInstance;
+
+@end
+
+
+NS_ASSUME_NONNULL_END
+
+
+@implementation RCInMemoryCachedObject
+
+- (instancetype)initWithCacheDurationInSeconds:(int)cacheDurationInSeconds {
+    return [self initWithCacheDurationInSeconds:cacheDurationInSeconds
+                                  lastUpdatedAt:nil];
+}
+
+- (instancetype)initWithCacheDurationInSeconds:(int)cacheDurationInSeconds
+                                 lastUpdatedAt:(NSDate *_Nullable)lastUpdatedAt {
+    if (self == [super init]) {
+        self.cacheDurationInSeconds = cacheDurationInSeconds;
+        self.lastUpdatedAt = lastUpdatedAt;
+    }
+    return self;
+}
+
+- (BOOL)isCacheStale {
+    if (self.lastUpdatedAt == nil) {
+        return YES;
+    }
+
+    NSTimeInterval timeSinceLastCheck = -1 * [self.lastUpdatedAt timeIntervalSinceNow];
+    return timeSinceLastCheck >= self.cacheDurationInSeconds;
+}
+
+- (void)clearCacheTimestamp {
+    self.lastUpdatedAt = nil;
+}
+
+- (void)clearCache {
+    [self clearCacheTimestamp];
+    self.cachedInstance = nil;
+}
+
+- (void)cacheInstance:(id)instance date:(NSDate *)date {
+    self.cachedInstance = instance;
+    self.lastUpdatedAt = date;
```

Comment:
Isn't this always `[NSDate date];`

Reply:
yes, currently, but I figured that there might be cases where `RCInMemoryCachedObjects` might get fetched before caching, and I didn't want to limit the possibility. 
Also, it seemed much easier for testing - otherwise you have to do a dance around NSDate to mock the return of date